dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -914,4 +916,84 @@ private static boolean isVarArgsInvocation(
     return TypesUtils.getArrayDepth(ElementUtils.getType(lastParamElt))
         != TypesUtils.getArrayDepth(lastArgType);
   }
+
+  /**
+   * Find the declaration of the receiver of a method call in a method tree.
```

Comment:
This says "the receiver", but the method body calls `tree.getParameters()` which is more than the receiver.

Reply:
I think there might be an ambiguity here due to my poorly-written documentation.

This method returns the declaration of the `receiver` of a method call that might appear in a method tree. An example might be:

```java
void foo(A a, B b) {
    System.out.println("irrelevant");
    a.invok();
}
```

If we pass the method tree `foo` and the receiver `JavaExpression a` to `getReceiverDeclarationInMethod`, it would look up the value of `a` in the body (where it is used, but not declared) or in the parameters of `foo` (where it is declared - `A a`).

Does this make more sense?